core/src/main/java/org/apache/james/core/quota/QuotaCurrentValue.java

```diff
@@ -0,0 +1,76 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import java.util.Objects;
+
+import org.apache.james.core.Username;
+
+public class QuotaCurrentValue {
+
+    public static QuotaCurrentValue of(QuotaComponent quotaComponent, Username identifier, QuotaType quotaType, long currentValue) {
```

Comment:
I think it does not fit. the domain part in QuotaRoot is optional while the identifier in QuotaCurrentValue always has this part

Reply:
Nope you are wrong here actually :) 

Look in UsersRepository code and configuration file... We have a parameter, `supportVirtualHosting`. If it's true we have domain names, if false we don't.

Reply:
I see that we still could use Username class that have the same logic as QuotaRoot in terms of username. Username class even provide more functions for username than QuotaRoot

Reply:
Not sure, but I just quickly check `com.linagora.tmail.team.TMailQuotaRootResolver`, Quotaroot is not only Username, It depends on which impl of `UserQuotaRootResolver` 
Example quotaRoot of TeamMailbox

```
def quotaRoot: QuotaRoot = QuotaRoot.quotaRoot(s"$TEAM_MAILBOX_NAMESPACE&${mailboxName.value.value}@${domain.asString()}", Some(domain).toJava)
```

Reply:
Yeah actually I am a little bit confused now. I found an example of quota root that looks strange in DefaultUserQuotaRootResolverTest, it is "private&benwa". Does anyone know what this mean?

Reply:
IMO `#private` is namespace

Reply:
I think the same as Rene.
`QuotaRoot` is a kind of ID for quota already.

> Username class even provide more functions for username than QuotaRoot

QuotaRoot exists for a reason, otherwise, the current quota code would use Username directly instead of QuotaRoot, no?

Reply:
what is namespace? :))

Reply:
> what is namespace? :))

https://www.rfc-editor.org/rfc/rfc2342.html

QUOTAROOT IMAP commands spec: https://www.rfc-editor.org/rfc/rfc2087

QUOTAROOT IMAP example: https://github.com/apache/james-project/blob/master/mpt/impl/imap-mailbox/core/src/main/resources/org/apache/james/imap/scripts/Quota.test

It looks like QuotaRoot is actually the id of quota, and is specially bound to the IMAP concept. As Rene said, I think that we should still respect the QuotaRoot here.

Reply:
If I use QuotaRoot, I will have to use something like "private&benwa" in which I do not really understand why we need the keywork private

Reply:
Forget the Quota Root actually :)